public/js/components/DataPropertyField.vue

```diff
@@ -0,0 +1,30 @@
+<template>
+  <label v-bind:class="{ invalid: !isValid }">
+    Data Property (supports
+    <a href="http://jmespath.org/tutorial.html" target="_blank">JMESPath</a>)
+    <input type="text" v-bind:name="name" v-model="model" v-bind:aria-invalid="!isValid" />
```

Comment:
The label should also be explicitly associated with the input element via a `for` attribute on the label and a matching `id` attribute on the input. Explicity label-control association is better supported by screen readers than implicit association (i.e. nesting the input inside the label).

Reply:
OK. How do you recommend I generate a unique ID to use for this? Everything the internet suggests looks like workarounds to me.

Reply:
In the last stream Suz generated a unique ID using the `tile.id`: https://github.com/noopkat/electric-io/blob/master/public/js/components/BaseCard.vue#L54

In the `DataPropertyField.vue` component, we could accept the `tile.id` as a prop `tileId`, and then use it in the input ID (e.g. `<input :id="'data-property-input-${tileId}'">`

Reply:
This is a very common problem in frameworks like Vue.js. One way I see recommended a lot for handling this in Vue is creating a [global mixin](https://vuejs.org/v2/guide/mixins.html#Global-Mixin) like suggested in this comment: https://github.com/vuejs/vue/issues/5886#issuecomment-308647738. Another option is using a third part package like [vue-uniq-ids](https://www.npmjs.com/package/vue-uniq-ids).

This might be a bit out of scope for this pull request, so feel free to ignore this. This is not a blocker for merging this pull request.